app/javascript/controllers/range_slider_controller.js

```diff
@@ -0,0 +1,175 @@
+    this.value = [
+      this.value[0],
+      this.clamp(next, this.value[0] + this.minDistanceValue, this.maxValue),
+    ]
+    this.render()
+  }
+
+  handlePointer(e, which) {
+    const rect = this.track.getBoundingClientRect()
+    const clientX = 'touches' in e ? e.touches[0].clientX : e.clientX
+    const percent = this.clamp(
+      ((clientX - rect.left) / rect.width) * 100,
+      0,
+      100
+    )
+    const raw = this.percentToValue(percent)
+    if (which === 'lo') this.setLo(raw)
+    else this.setHi(raw)
+  }
+
+  onTrackDown = e => {
+    e.preventDefault()
+    const rect = this.track.getBoundingClientRect()
+    const clientX = 'touches' in e ? e.touches[0].clientX : e.clientX
+    const percent = this.clamp(
+      ((clientX - rect.left) / rect.width) * 100,
+      0,
+      100
+    )
+    const raw = this.percentToValue(percent)
+    const dLo = Math.abs(raw - this.value[0])
+    const dHi = Math.abs(raw - this.value[1])
+    const which = (dLo === dHi ? raw < this.value[0] : dLo < dHi) ? 'lo' : 'hi'
+    which === 'lo' ? this.setLo(raw) : this.setHi(raw)
+    this.activeThumb = which
+    document.addEventListener('mousemove', this.onMove)
+    document.addEventListener('mouseup', this.onUp)
+    document.addEventListener('touchmove', this.onMove, { passive: false })
+    document.addEventListener('touchend', this.onUp)
+  }
+
+  onThumbDown = which => e => {
+    e.preventDefault()
+    this.activeThumb = which
+    document.addEventListener('mousemove', this.onMove)
+    document.addEventListener('mouseup', this.onUp)
+    document.addEventListener('touchmove', this.onMove, { passive: false })
+    document.addEventListener('touchend', this.onUp)
+  }
+
+  onMove = e => {
+    if (!this.activeThumb) return
+    this.handlePointer(e, this.activeThumb)
+  }
+
+  onUp = () => {
+    this.activeThumb = null
+    document.removeEventListener('mousemove', this.onMove)
+    document.removeEventListener('mouseup', this.onUp)
+    document.removeEventListener('touchmove', this.onMove)
+    document.removeEventListener('touchend', this.onUp)
+  }
+
+  onInputChange = which => e => {
+    const val = parseInt(e.target.value, 10)
+    if (isNaN(val)) return
+    if (which === 'lo') this.setLo(val)
+    else this.setHi(val)
+  }
+
+  render() {
+    if (!this.container) {
+      this.container = document.createElement('div')
+      this.container.className = 'w-full select-none'
+      this.container.innerHTML = `
+        <div class="rs-inputs">
+          <input type="number" class="rs-input" name="${this.keyBaseValue}_greater_than" />
+          <input type="number" class="rs-input" name="${this.keyBaseValue}_less_than" />
+        </div>
+        <div class="rs-track">
+          <div class="rs-bar"></div>
+          <div class="rs-range text-blue-600"></div>
+          <button type="button" class="rs-thumb text-blue-600"></button>
+          <button type="button" class="rs-thumb text-blue-600"></button>
+        </div>`
+
```

Comment:
Hmm - normally you would use targets with Stimulus: https://stimulus.hotwired.dev/reference/targets